Database

Lukease opened the pull request.

Commit:
initial commit

Commit:
create new type: one-recipe,
create new file: recipe-database_utils, recipesIngredients,recipesIngredients-database_utils,
rename directory to http-client.

Commit:
remove async await init ,
edit functions in http-client,
create new class error-menu

Commit:
remove enter

Commit:
remove enter

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+import {renderProduct} from '../products_utils'
```
add spaces between `{}` in the entire project

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+import {renderProduct} from '../products_utils'
+import {ingredientData} from '../type/ingredient-data'
```
this is invalid import, add index.ts and export things.
Imports should be one level deep eg.:
```
import { ingredientData } from '../type'
```

Also file name should be probably `types`

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+import {ingredientData} from '../type/ingredient-data'
+
+export const getAllIngredients = async (ingredientsArray: Array<ingredientData>) => {
+    const loader = $('<div>').addClass('products__loader').appendTo($('.products'))
```
It's a good practice to chain operations in the given way:
```
const loader = $('<div>')
     .addClass('products__loader')
     .appendTo($('.products'))
```

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+
+export const getAllIngredients = async (ingredientsArray: Array<ingredientData>) => {
+    const loader = $('<div>').addClass('products__loader').appendTo($('.products'))
+    const url = 'http://localhost:3001/ingredients'
```
you can create a util function to append only the last part to the url. Localhost is mandatory so, you shouldn't repeat it multiple times

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+    const url = 'http://localhost:3001/ingredients'
+
+    await fetch(url, {
+        method: 'GET'
```
GET is default value, you can use shorthand:
```
fetch(url)
```

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+        .then(async res => {
+            ingredientsArray = ingredientsArray.concat(JSON.parse(await res.text()))
+        })
+        .catch(error => {
```
it will never go to the catch. I showed how to resolve it this morning on the whiteboard

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+            ingredientsArray = ingredientsArray.concat(JSON.parse(await res.text()))
+        })
+        .catch(error => {
+            $('<div>').addClass('products__error').text(error).appendTo($('.loader-container')).css('color', 'red').css('font-size', '30px')
```
add enter for chaining

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+            loader.remove()
+        })
+
+    ingredientsArray.forEach((object) => {
```
no need for `()`

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+
+    ingredientsArray.forEach((object) => {
+        renderProduct(object.name, object.ingredientsId)
+        $('.selected-product__text').attr('readOnly', 'true').css('text-align', 'center')
```
enters

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+}
+
+export const deleteIngredient = (ingredientId: string) => {
+    const url: string = `http://localhost:3001/ingredients?ingredientsId=${ingredientId}`
```
aa

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+    return fetch(url, {
+        method: 'DELETE',
+    })
+        .then((response) => response.json())
```
aa

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+            newIngredientName: newName
+        })
+    })
+        .then((response) => response.json())
```
aa

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+}
+
+export const editIngredient = async (ingredientId: number, newName: string) => {
+    const url = 'http://localhost:3001/ingredients'
```
aa

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+}
+
+export const addNewIngredient = async (ingredientName: string) => {
+    const url = `http://localhost:3001/ingredients?ingredientName=${ingredientName}`
```
aa

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+    return fetch(url, {
+        method: 'POST',
+    })
+        .then((response) => response.json())
```
aa

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -0,0 +1,72 @@
+        .catch(error => {
+            alert(`cant add ${ingredientName}` + error)
+        })
+}
```
given all the comments above, you can update every file to follow these rules

Commit:
create new file fetch-config.ts, add base url to all methods in http-client,
create index.ts in types file, rename file from 'type' to 'types'
remove method: 'GET' from all get method in http-client,

jpudysz commented on fetch-config.ts:
```diff
@@ -0,0 +1,2 @@
+export const baseUrl = 'http://localhost:3001/'
```
I can't see additional axios config

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -1,48 +1,51 @@
-    await fetch(url, {
-        method: 'GET'
-    })
+    await fetch(`${baseUrl}ingredients`)
```
it can be done better with axios global config

jpudysz commented on http-client/ingredients-database_utils.ts:
```diff
@@ -1,48 +1,51 @@
         })
 
-    ingredientsArray.forEach((object) => {
+    ingredientsArray.forEach( object => {
```
remove space, also object is too generic, use more descriptive name

jpudysz commented on http-client/recipes-database_utils.ts:
```diff
@@ -1,63 +1,63 @@
-        .then(async res => {
-            recipesArray = recipesArray.concat(JSON.parse(await res.text()))
+    return fetch(`${baseUrl}recipes/by-all`)
+        .then(async response => {
```
you can write it better:
```
fetch(...)
   .then(response => response.text())
   .then(recipes => recipesArray.concat(recipes))
```

jpudysz commented on http-client/recipes-database_utils.ts:
```diff
@@ -1,63 +1,63 @@
+                .text(error).appendTo($('.recipes'))
         })
-        .finally(() => {
+        .then(() => {
```
finally

jpudysz commented on http-client/recipesIngredients-database_utils.ts:
```diff
@@ -53,43 +50,41 @@ export const addNewRecipesIngredients = (ingredientId: number, recipesId: number
-    return fetch(url, {
-        method: 'GET',
-    })
+    return fetch(`${baseUrl}recipesIngredients/by-ingredients?ingredientsId=${ingredientsId}`)
```
use axios global config and fix these urls

Lukease replied:
i cant config axios, every single time doesnt work

jpudysz replied:
find a solution :) you can google it

jpudysz commented on main.ts:
```diff
@@ -1,26 +1,47 @@
+import { renderProduct } from './products_utils'
+import { renderRecipesMenu, renderSaveRecipes } from './recipes_utils'
+import { renderMishMashChoice } from './mish_mash_utils'
+import { getAllIngredients } from './http-client/ingredients-database_utils'
```
bad import

jpudysz commented on mish_mash_utils.ts:
```diff
@@ -1,35 +1,55 @@
-import {OneRecipe} from './type/one-recipe'
+import { addToRecipe, removeFromRecipe } from './recipes_utils'
+import { ingredientData, OneRecipe } from './types'
+import { getAllIngredients } from './http-client/ingredients-database_utils'
```
bad imports, add index.ts

jpudysz commented on recipes_utils.ts:
```diff
@@ -1,13 +1,11 @@
-import {ingredient} from './type/ingredients'
-import {getAllIngredients} from './http-client/ingredients-database_utils'
-import {ingredientData} from './type/ingredient-data'
+import { addNewRecipe, deleteRecipe, getAllRecipes, getOneRecipe } from './http-client/recipes-database_utils'
```
bad imports
